src/com/company/CricketMatch.java

```diff
@@ -0,0 +1,221 @@
+                if(generatedOutput.equals("Wicket"))
+                {
+                    wicketGone();
+                    if(battingTeam.getWickets() >= 10){
+                        return;
+                    }
+                    battingTeam.addNumberOfBallPlayed(1);
+                    strikeBowler.addNoOfBallsBowled(1);
+                }
+                else if((generatedOutput.equals("Wide")) || (generatedOutput.equals("5")))
+                {
+                    System.out.println("Wide");
+                    if(generatedOutput.equals("Wide"))
+                    {
+                        battingTeam.addRun(1);
+                        battingTeam.addWideRun(1);
+                        strikeBowler.addGivenRun(1);
+                    }
+                    else
+                    {
+                        battingTeam.addRun(5);
+                        battingTeam.addWideRun(5);
+                        strikeBowler.addGivenRun(5);
+                    }
+                    j--;
+                }
+                else if(generatedOutput.equals("NoBall"))
+                {
+                    System.out.print("No Ball and ");
+                    battingTeam.addRun(1);
+                    battingTeam.addNoBallRun(1);
+                    strikeBowler.addGivenRun(1);
+                    int noBallRun = (int)(Math.random() * 6);
+                    if(noBallRun == 5)noBallRun = 6; // No Ball And Wide + 4, Not happen.
+                    updateScoreBoard(noBallRun);
+                    printCommentary(noBallRun);
+                    j--;
+                }
+                else
+                {
+                    int runCount = Integer.parseInt(generatedOutput);
+                    updateScoreBoard(runCount);
+                    printCommentary(runCount);
+                    battingTeam.addNumberOfBallPlayed(1);
+                    strikeBowler.addNoOfBallsBowled(1);
+                }
+                if((inning_no == 2) && (bowlingTeam.getRun() < battingTeam.getRun()))
+                {
+                    return;
+                }
+            }
+            swapPlayer();
+        }
+        System.out.println("\n" + battingTeam.getName() +" Inning was over");
+        System.out.println("------------------------------------------------------------------------------------------------------");
+    }
+
+    public void startGame(int totalOver)
+    {
+        System.out.println("New Game Start!");
+
+        innings(totalOver,1);
+        swapTeam();
+        innings(totalOver,2);
+    }
+
+    public void winningTeam()
+    {
+        System.out.println("\n-----------------------------------------------------------------------------");
+
+        if(firstBattingTeam.getRun() == secondBattingTeam.getRun())
+        {
+            System.out.println("Match Tie");
+        }
+        else if(firstBattingTeam.getRun() > secondBattingTeam.getRun())
```

Comment:
firstBattingTeam and secondBattingTeam are redundant variables. 
after a match is played, battingTeam will always be the team that batted in second innings and bowlingTeam will be the one that bowled in second innings.
so if battingTeam won, that means if won by some number of wickets , and if bowlingTeam won that means it won by some number of runs